lib/mqtt/mqtt.cc

```diff
@@ -954,3 +943,42 @@ int mqtt_run(Timeout *t, SObj mqttHandle)
 		  return 0;
 	  });
 }
+
+int mqtt_generate_client_id(char *buffer, size_t length)
+{
+	/**
+	 * Note from the MQTT 3.1.1 spec:
+	 * The Server MUST allow ClientIds which are between 1 and 23 UTF-8 encoded
+	 * bytes in length, and that contain only the characters
+	 * "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	 *
+	 * Note from us:
+	 * UTF-8 encoding of 0-9, a-z, A-Z, is 1 Byte per character, so we should be
+	 * able to do up to a length of 22 characters + zero byte.
+	 */
+	const char characters[] =
```

Comment:
If you make this a std::string_view, you can use .size() instead of sizeof calculations.

Reply:
For some reason I'm hitting an unrelated bug with std::string_view, so I would rather leave this for future works.